### PR TITLE
Sync command-reference.md

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
And since I was at it, I changed the generated markdown file to a bit cleaner and consistent.

### CODE REVIEW

**Comment by deivid-rodriguez** [comment]
I ended up adding some general maintenance in this PR. Proposed changes include:

*  Sync command reference with its sources and make sure doesn't fall out of sync again.
* Upgrade command reference to use the latest rubygems version. It was previously generated with v3.0.2, so a bit outdaded.
* Cleanup generated markdown file to not include whitespace only lines, or duplicated blank lines, and ensure it stays like that.

**Comment by deivid-rodriguez** [comment]
I'll merge this in a couple of days if I don't get any reviews.

File: command-reference.erb [hunk 1]
```diff
@@ -10,31 +10,28 @@ next: /rubygems-org-api
...
+<%- commands.each do |name, command| -%><%= "* [gem #{name}](#gem-#{name})\n" %><%- end -%>
+<%- commands.each do |name, command| -%>
 
-<% commands.each do |name, command| %>
 ## gem <%= name %>
 
 <%= htmlify(command.summary) %>
 
-### Usage
+### Usage:
```
Comment (igneus) [inline command-reference.erb:20]: I would rather remove the colon everywhere than add it everywhere. But that's just a matter of taste.

Comment (deivid-rodriguez) [reply command-reference.erb]: I agree it doesn't add much, updated!

**Pull request merged**